swim/disseminator.go

```diff
@@ -245,3 +248,57 @@ func (d *disseminator) ChangesCount() int {
 	d.RUnlock()
 	return c
 }
+
+// tryStartReverseFullSync fires a goroutine that performs a full sync, with a
+// maximum of five concurrent goroutines at the same time. If five are already
+// running no action is undertaken.
+func (d *disseminator) tryStartReverseFullSync(target string, timeout time.Duration) {
+	if d.reverseFullSyncJobs == nil {
+		d.reverseFullSyncJobs = make(chan struct{}, d.node.maxReverseFullSyncJobs)
+	}
+
+	// occupy a job, return if none are available
+	select {
+	case d.reverseFullSyncJobs <- struct{}{}:
+		// continue if job is available
+	default:
+		d.logger.WithFields(log.Fields{
+			"remote": target,
+		}).Info("omit bidirectional full sync, too many already running")
+
+		d.node.emit(OmitReverseFullSyncEvent{Target: target})
+		return
+	}
+
+	// start reverse full sync
+	go func() {
+		d.reverseFullSync(target, timeout)
+
+		// create a new vacancy when the job is done
+		<-d.reverseFullSyncJobs
+	}()
+}
+
+// reverseFullSync is the second part of a bidirectional full sync. It ensures
```

Comment:
Where is the first part? If you are documenting the function, add context either to the header of the file, or somewhere close.